include/fcl/narrowphase/detail/primitive_shape_algorithm/halfspace-inl.h

```diff
@@ -371,7 +371,7 @@ bool cylinderHalfspaceIntersect(const Cylinder<S>& s1, const Transform3<S>& tf1,
   Vector3<S> dir_z = R.col(2);
   S cosa = dir_z.dot(new_s2.n);
 
-  if(cosa < halfspaceIntersectTolerance<S>())
+  if(std::abs(cosa) < halfspaceIntersectTolerance<S>())
```

Comment:
I'm putting this comment here because github won't let me put it where I *want* to.

On line 394 we have the line of code:

```C++
if (std::abs(cosa + 1) < halfspaceIntersectTolerance<S>() || std::abs(cosa - 1) < halfspaceIntersectTolerance<S>())
```

It could be replaced with:

```C++
if (std::abs(cosa) - 1 < halfspaceIntersectTolerance<S>())
```

I figured, as long as we're in this code, we could patch this as well (although this isn't a correctness issue, merely an efficiency/compactness issue).